src/bam/mod.rs

```diff
@@ -808,6 +808,8 @@ mod tests {
         let mut rec = record::Record::new();
         rec.set_reverse();
         rec.set(names[0], &cigars[0], seqs[0], quals[0]);
+        // note: this segfaults if you push_aux() before set()
+        //       because set() obliterates aux
```

Comment:
Maybe you can also add that to the corresponding docstrings: "push_aux may be never called before set".